src/Workspaces/Remote/Core/Services/SolutionCreator.cs

```diff
@@ -497,5 +507,49 @@ private Project AddDocument(Project project, DocumentInfo documentInfo)
         {
             return project.Solution.AddDocument(documentInfo).GetProject(project.Id);
         }
+
+        private CompilationOptions FixUpCompilationOptions(ProjectInfo.ProjectAttributes info, CompilationOptions compilationOptions)
+        {
+            return compilationOptions.WithXmlReferenceResolver(new XmlFileResolver(GetXmlResolverBasePath(info.FilePath)))
+                                     .WithStrongNameProvider(new DesktopStrongNameProvider(GetStrongNameKeyPaths(info)));
+        }
+
+        private static string GetXmlResolverBasePath(string filePath)
+        {
+            if (!PathUtilities.IsAbsolute(filePath))
```

Comment:
That's a bit unfortunate. Then we should review that all use-sites are hardened against bad paths.

For example,
http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Next/Options/EditorConfigDocumentOptionsProvider.cs,99
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.AbstractGlobalSuppressMessageCodeAction.cs,42